bumpversion/cli.py

```diff
@@ -89,230 +129,215 @@ def split_args_in_optional_and_positional(args):
 
+def _setup_logging(show_list, verbose):
     logformatter = logging.Formatter("%(message)s")
-
-    if len(logger_list.handlers) == 0:
+    if not logger_list.handlers:
         ch2 = logging.StreamHandler(sys.stdout)
         ch2.setFormatter(logformatter)
         logger_list.addHandler(ch2)
-
-    if known_args.list:
+    if show_list:
         logger_list.setLevel(logging.DEBUG)
-
     try:
-        log_level = [logging.WARNING, logging.INFO, logging.DEBUG][known_args.verbose]
+        log_level = [logging.WARNING, logging.INFO, logging.DEBUG][verbose]
     except IndexError:
         log_level = logging.DEBUG
-
     root_logger = logging.getLogger('')
     root_logger.setLevel(log_level)
+    logger.debug("Starting %s", DESCRIPTION)
 
-    logger.debug("Starting {}".format(DESCRIPTION))
 
-    defaults = {}
-    vcs_info = {}
-
-    for vcs in VCS:
+def _determine_vcs_usability(possible_vcses, vcs_info):
+    for vcs in possible_vcses:
         if vcs.is_usable():
             vcs_info.update(vcs.latest_tag_info())
 
+
+def _determine_current_version(vcs_info, defaults):
     if "current_version" in vcs_info:
         defaults["current_version"] = vcs_info["current_version"]
 
-    explicit_config = hasattr(known_args, "config_file")
 
+def _determine_config_file(explicit_config):
     if explicit_config:
-        config_file = known_args.config_file
-    elif not os.path.exists(".bumpversion.cfg") and os.path.exists("setup.cfg"):
-        config_file = "setup.cfg"
-    else:
-        config_file = ".bumpversion.cfg"
+        return explicit_config
+    if not os.path.exists(".bumpversion.cfg") and os.path.exists("setup.cfg"):
+        return "setup.cfg"
+    return ".bumpversion.cfg"
+
 
+def _load_configuration(config_file, explicit_config, files, defaults, part_configs):
     # setup.cfg supports interpolation - for compatibility we must do the same.
     if os.path.basename(config_file) == "setup.cfg":
         config = ConfigParser("")
     else:
         config = RawConfigParser("")
-
     # don't transform keys to lowercase (which would be the default)
     config.optionxform = lambda option: option
-
     config.add_section("bumpversion")
-
     config_file_exists = os.path.exists(config_file)
 
-    part_configs = {}
-
-    files = []
-
-    if config_file_exists:
+    if not config_file_exists:
+        message = "Could not read config file at {}".format(config_file)
+        if explicit_config:
+            raise argparse.ArgumentTypeError(message)
+        logger.info(message)
+        return config, config_file_exists
 
-        logger.info("Reading config file {}:".format(config_file))
-        # TODO: this is a DEBUG level log
-        logger.info(io.open(config_file, "rt", encoding="utf-8").read())
+    logger.info("Reading config file %s:", config_file)
+    # TODO: this is a DEBUG level log
+    logger.info(io.open(config_file, "rt", encoding="utf-8").read())
```

Comment:
Should we also improve this while we're at it?

```suggestion
    with io.open(config_file, "rt", encoding="utf-8") as config_fp:
        logger.info(config_fp.read())
```

Reply:
OK to me!